etl/steps/data/garden/maternal_mortality/2024-07-08/maternal_mortality.py

```diff
@@ -0,0 +1,225 @@
+    tb_wpp = pr.merge(tb_wpp_births, tb_wpp_fm_pop, on=["country", "year"], how="left")
+
+    # combine with who mortality and calculate maternal mortality ratio and rate
+    tb_calc_mm = pr.merge(tb_wpp, tb_who_mortality, on=["country", "year"], how="outer")
+    tb_calc_mm["mmr"] = (tb_calc_mm["maternal_deaths"] / tb_calc_mm["live_births"]) * 100_000
+    tb_calc_mm["mm_rate"] = (tb_calc_mm["maternal_deaths"] / tb_calc_mm["female_population"]) * 100_000
+
+    # combine with Gapminder data - using WHO/ UN WPP data where available
+    tb_gm["maternal_deaths"] = tb_gm["maternal_deaths"].round().astype("UInt32")
+    tb_who_gm = combine_two_overlapping_dataframes(tb_calc_mm, tb_gm, index_columns=["country", "year"])
+
+    # calculate mmr and mm rate if NA from Gapminder deaths with UN WPP data
+    tb_who_gm["mmr"] = tb_who_gm["mmr"].combine_first(tb_who_gm["maternal_deaths"] / tb_who_gm["live_births"] * 100_000)
+    tb_who_gm["mm_rate"] = tb_who_gm["mm_rate"].combine_first(
+        tb_who_gm["maternal_deaths"] / tb_who_gm["female_population"] * 100_000
+    )
+
+    # join the two tables
+    # first - rename columns so they have the same names
+    tb_un = tb_un.rename(columns={"births": "live_births"})
+
+    # combine UN MMEIG with WHO/UN/GM using UN MMEIG preferentially
+    tb = combine_two_overlapping_dataframes(tb_un, tb_who_gm, index_columns=["country", "year"])
+
+    # remove all rows that don't have any data on maternal deaths, mmr or mm rate
+    tb = tb.dropna(subset=["maternal_deaths", "mmr", "mm_rate"], how="all")
+
+    # calculate regional aggregates - population is needed for filtering out all regions that are not sufficiently covered by our data
+    tb = geo.add_population_to_table(tb, ds_pop)
+
+    aggr = {"maternal_deaths": "sum", "live_births": "sum", "female_population": "sum", "population": "sum"}
+
+    tb = geo.add_regions_to_table(
+        tb=tb,
+        regions=REGIONS,
+        ds_regions=ds_regions,
+        ds_income_groups=ds_income,
+        aggregations=aggr,
+        num_allowed_nans_per_year=0,
+    )
+
+    # remove all regions that are less than 90% covered by our data
+    tb = check_region_share_population(tb, REGIONS, ds_pop, 0.9)
+
+    # calculate aggregated maternal mortality ratio and rate for regions
+    tb["mmr"] = tb.apply(lambda x: calc_mmr(x), axis=1)
+    tb["mm_rate"] = tb.apply(lambda x: calc_mmrate(x), axis=1)
+
+    # drop all columns that are not 1) long run or 2) not related to maternal mortality
+    cols_to_keep = ["country", "year", "maternal_deaths", "mmr", "live_births", "mm_rate"]
+    tb = tb[cols_to_keep]
+
+    # drop rows where there is no data for maternal mortality indicators (introduced by aggregating regions)
+    tb = tb.dropna(subset=["maternal_deaths", "mmr", "mm_rate"], how="all")
+
+    # fix dtypes (coerce errors since NAs are not accepted otherwise)
+    tb["maternal_deaths"] = (
+        pd.to_numeric(tb["maternal_deaths"], errors="coerce")
```

Comment:
Is `pr.to_numeric()` not possible? I'm not sure if it's included or not.

Reply:
I tried it and it didn't work :( I also looked for equivalent functions (something like tb.to_numeric()), but that also doesn't exist. Maybe we should add it?

Reply:
Oh sorry I missed this! It looks like it is figured out now?